src/main/java/com/netgrif/application/engine/workflow/web/PublicTaskController.java

```diff
@@ -186,16 +186,15 @@ public ResponseEntity<Resource> getNamedFile(@PathVariable("id") String taskId,
         return super.getNamedFile(taskId, fieldId, name);
     }
 
-    @Override
     @Operation(summary = "Remove file from tasks file list field value",
             description = "Caller must be assigned to the task, or must be an ADMIN")
     @DeleteMapping(value = "/{id}/file/{field}/{name}", produces = MediaTypes.HAL_JSON_VALUE)
     @ApiResponses(value = {
             @ApiResponse(responseCode = "200", description = "OK"),
             @ApiResponse(responseCode = "403", description = "Caller doesn't fulfill the authorisation requirements"),
     })
-    public EntityModel<EventOutcomeWithMessage> deleteNamedFile(@PathVariable("id") String taskId, @PathVariable("field") String fieldId, @PathVariable("name") String name) {
-        return super.deleteNamedFile(taskId, fieldId, name);
+    public EntityModel<EventOutcomeWithMessage> deleteNamedFile(@PathVariable("id") String taskId, @PathVariable("field") String fieldId, @PathVariable("name") String name, @RequestParam("parentTaskId") String parentTaskId) {
```

Comment:
taskId??

Reply:
Here the taskId is used for authorisation and parentTaskId is either equals to taskId or in case of taskRef it equals to the referenced task ID. If I use only the taskId, there are scenarios, where this function does not work and throws exception (e.g. if the referenced task is from another case).

Added missing authorisation in this case.

Reply:
If `parentTaskId` parameter is **referenced** task in use of taskRef construct, shouldn't be better to call it referencedTaskId. Parent task signifies more a task with the taskRef field, not the taskRef value. 
This is only my opinion ... 🤔

Reply:
Yes, you are right, but then the whole construction on frontend should be refactored too. I experienced, the name `parentTaskId` is used on frontend since the taskRef handling was implemented, I just used the existing naming convention on backend. 

Please consider, that the `parentTaskId` parameter holds the ID of task, where the dataField is referenced (if data field is directly in the task, then the `parentTaskId` and `taskId` will be the same, if the dataRef is in task that is referenced through taskRef, `parentTaskId` holds the referenced task). That's how it works on frontend too, when resolving taskId for task panel data.

If you wish, I can refactor it.

Reply:
I understand now. Maybe this is a topic for further discussion.